File: orig/Game/src/Background.java
import javax.microedition.lcdui.*;

class Background extends GameObject {
    private final int tile_size = 30;
    private int base_y = 0;

    public int speed = 2;

    public Background() {}

    public void update() {
        base_y = (base_y + speed) % (tile_size * 2);
    }

    public void paint(Graphics g) {
	GameCanvas c = canvas;

        for (int y = -2; y <= c.getHeight() / tile_size; y++) {
            for (int x = 0; x <= c.getHeight() / tile_size; x++) {
                g.setColor(((x+y) % 2 != 0) ? 0xFFFFFF : 0xAAAAAA);
                g.fillRect(x * tile_size,
                           y * tile_size + base_y,
                           tile_size, tile_size);
            }
        }
    }
}

File: js/background.js
// Background.java port — scrolling checkerboard

import { GameObject } from './gameObject.js';

export class Background extends GameObject {
    constructor(game) {
        super(game);
        this.tileSize = 30;
        this.baseY = 0;
        this.speed = 2;
    }

    update() {
        this.baseY = (this.baseY + this.speed) % (this.tileSize * 2);
    }

    paint(g) {
        const W = this.game.WIDTH;
        const H = this.game.HEIGHT;
        for (let y = -2; y <= H / this.tileSize; y++) {
            for (let x = 0; x <= H / this.tileSize; x++) {
                g.setColor(((x + y) % 2 !== 0) ? 0xFFFFFF : 0xAAAAAA);
                g.fillRect(
                    x * this.tileSize,
                    y * this.tileSize + this.baseY,
                    this.tileSize, this.tileSize
                );
            }
        }
    }
}
